SquareLine/gen.py

```diff
@@ -0,0 +1,231 @@
+    check_json_key(manifest, "version")
+    v = manifest["version"].split('.')
+    return "_v" + v[0] + "_" + v[1] + "_" + v[2]
+
+def create_slb_file(output, output_filename, manifest):
+    check_json_key(manifest, "name")
+    check_json_key(manifest, "version")
+    check_json_key(manifest, "mcu")
+    check_json_key(manifest, "screen_width")
+    check_json_key(manifest, "screen_height")
+    check_json_key(manifest, "screen_color_swap")
+    check_json_key(manifest, "short_description")
+    check_json_key(manifest, "long_description")
+
+    SLB_FILE["version"] = manifest["version"]
+    SLB_FILE["title"] = manifest["name"]
+    SLB_FILE["keywords"] += ", " + manifest["name"] + ", " + manifest["mcu"]
+    SLB_FILE["width"] = manifest["screen_width"]
+    SLB_FILE["width_min"] = manifest["screen_width"]
+    SLB_FILE["width_max"] = manifest["screen_width"]
+    SLB_FILE["height"] = manifest["screen_height"]
+    SLB_FILE["height_min"] = manifest["screen_height"]
+    SLB_FILE["height_max"] = manifest["screen_height"]
+    if manifest["screen_color_swap"]:
+        SLB_FILE["color_depth"] = "16 sw"
+    else:
+        SLB_FILE["color_depth"] = "16"
+    SLB_FILE["short_description"] = manifest["short_description"]
+    SLB_FILE["long_description"] = manifest["long_description"]
+
+    out_slb_json = json.dumps(SLB_FILE, indent=4)
+    f = open(output + "/" + output_filename + ".slb", "w")
+    f.write(out_slb_json)
+    f.close()
+
+    print(f"  File {output_filename}.slb created.")
+
+
+# Process of the board generating
+def process_board(board_name, output, dir):
+    global placeholders
+    print(f"Processing board: {dir}")
+
+    path = dir + "/" + BOARD_GENFILE
+    if os.path.exists(path):
+        # Parse manifest file
+        f = open(path, "r")
+        manifest = json.loads(f.read())
+        f.close()
+
+        # Get string version for file names
+        file_ver = get_file_version(manifest)
+        output_filename = board_name + file_ver
+
+        check_json_key(manifest, "placeholders")
+        placeholders = manifest["placeholders"]
+
+        # Copy common files
+        copy_directory(COMMON_DIR, output + "/" + SQUARELINE_OUTDIR)
+        # Copy specific board files
+        for file in BOARD_SPECIFIC_FILES:
+            copy_file(dir + "/" + file, output + "/" + SQUARELINE_OUTDIR + "/" + file)
+        # Copy image
+        copy_file(dir + "/image.png", output + "/" + output_filename + ".png", 0)
+
+        create_slb_file(output, output_filename, manifest)
+        make_zip(output_filename, output)
+        remove_folder(output + "/" + SQUARELINE_OUTDIR)
+    else:
+        print_error(f"ERROR: File '{path}' is not exists!")
+        os._exit(1)
+
+# Print usage
+def print_help():
+    print("Not enough arguments! Usage:")
+    print(" gen.py out board")
+    print("\tout: path to the generated output")
+    print("\tboard (optional): generate only specific board")
+
+
+def main():
+    if len(sys.argv) >= 2:
```

Comment:
Minor recommendation, please consider using `argparse` for parsing the command line arguments. You can check most of IDF python tools for examples.